File: 0012-Integer to Roman/0012-Integer to Roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<int> radix = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        vector<string> symbol = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string result = "";
        for (int i = 0; num > 0; ++i) {
            int count = num / radix[i];
            while (count) {
                result += symbol[i];
                --count;
            }
            
            num %= radix[i];
        }
        
        return result;
    }
};

File: 0001-0100/0012-Integer to Roman/0012-Integer to Roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        radix = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        symbol = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        result = []
        for r, s in zip(radix, symbol):            
            result.append(s * (num // r))
            num %= r
            if num == 0:
                break
        return ''.join(result)
